nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -412,6 +415,26 @@ protected void buildPartialRow(Schema schema, PartialRow row, Record record, Lis
         }
     }
 
+    /**
+     * Get java.sql.Date from Record Field Value with optional parsing when input value is a String
+     *
+     * @param value Record Field Value
+     * @param recordFieldName Record Field Name
+     * @return Date object or null when value is null
+     */
+    private Date getDate(final Object value, final String recordFieldName) {
+        return DataTypeUtils.toDate(value, () -> getDateFormat(), recordFieldName);
+    }
+
+    /**
+     * Get Date Format using Date Record Field default pattern and system time zone to avoid unnecessary conversion
+     *
+     * @return Date Format used to parsing date fields
+     */
+    private DateFormat getDateFormat() {
+        return new SimpleDateFormat(RecordFieldType.DATE.getDefaultFormat());
```

Comment:
Should this use `fieldType.getFormat()` similar to the handling for Timestamp/UNIXTIME_MICROS above?

Reply:
Thanks for the suggestion @granthenke, that seems like a helpful improvement as opposed to hard-coded default format of `yyyy-MM-dd` from `RecordFieldType.DATE.getDefaultFormat()`.  I updated the methods to accept the format from `DataType.getFormat()`.